src/main.cpp

```diff
@@ -149,6 +151,7 @@ int main(int argc, char *const argv[]) {
 
     if (program != nullptr && ::errorCount() == 0) {
         auto original = program;
+        pruner_conf.err_type = classify_bug(options.file, pruner_conf);
```

Comment:
We have this weird mix where we set both set a value in the conf from a return and in the function itself. This is confusing. Let's just return a struct which we use to set the values (and only set these at the beginning). Instead of classify_bug this will be `get_exit_info`, which calls classify_bug as a subroutine.

Reply:
But we need to know the type of the bug before we can get the exit info right?(since we get the exit code from diff binaries) So how can classify_bug be called from inside get_exit_info?

Reply:
Okay I think the code is a little mixed up. Classify bug should only be called once at the beginning and it needs some specific logic.

So if we use the conf we already know the exit code and which binary to use. For crash bugs we just need to classify. If we do not have a conf the user must supply either the compiler or the validation binary and the we retrieve the details.
`pruner_conf.err_type` and `pruner_conf.err_string` should only be set once.

 But we still need to retrieve that info every time we call `get_exit_info` (to check whether the class of bug is still the same). This also means checking the string every time.